framework/wazuh/core/indexer/bulk.py

```diff
@@ -0,0 +1,191 @@
+from typing import Optional, Dict, Any, List, Protocol, Coroutine
+from enum import Enum
+
+from opensearchpy import AsyncOpenSearch
+
+
+class BulkAction(str, Enum):
+    """Enum representing bulk actions for OpenSearch.
+
+    Attributes
+    ----------
+    CREATE : str
+        Action to create a new document.
+    CREATE_OR_UPDATE : str
+        Action to create a new document or update an existing one.
+    UPDATE : str
+        Action to update an existing document.
+    DELETE : str
+        Action to delete an existing document.
+    """
+    CREATE = "create"
+    CREATE_OR_UPDATE = "index"
+    UPDATE = "update"
+    DELETE = "delete"
+
+
+class BulkMetadata:
+    """Class to handle metadata for a bulk action in OpenSearch.
+
+    Parameters
+    ----------
+    index : str
+        The name of the index in OpenSearch.
+    doc_id : Optional[str]
+        The ID of the document in OpenSearch. Can be None for certain actions.
+    action : BulkAction
+        The type of bulk action to perform.
+    """
+    def __init__(self, index: str, doc_id: Optional[str], action: BulkAction):
+        self.index = index
+        self.doc_id = doc_id
+        self.action = action
+
+    def decode(self) -> Dict[str, Dict[str, str]]:
+        """Decode metadata into a dictionary format for the OpenSearch bulk API.
+
+        Returns
+        -------
+        Dict[str, Dict[str, str]]
+            The metadata in a dictionary format.
+        """
+        return {f'{self.action}': {"_index": self.index, "_id": self.doc_id}}
```

Comment:
```suggestion
        return {str(self.action: {'_index': self.index, '_id': self.doc_id}}
```

Reply:
Fixed in https://github.com/wazuh/wazuh/commit/47e4508357e0fb53f499847d6a791a0f4c86f15b.